diff -r ac8dfc32c6f5 -r 92bce7d0bc64 src/XmlRpcProxyGen.cs
--- a/src/XmlRpcProxyGen.cs	Sat Mar 03 13:53:52 2012 +0000
+++ b/src/XmlRpcProxyGen.cs	Sat Mar 03 13:54:32 2012 +0000
@@ -30,6 +30,24 @@
   using System.Reflection;
   using System.Reflection.Emit;
 
+  public sealed class XmlRpcProxyAssemblyNameAttribute : Attribute
+  {
+    private readonly string mAssemblyName;
+
+    public XmlRpcProxyAssemblyNameAttribute(string assemblyName)
+    {
+      mAssemblyName = assemblyName;
+    }
+
+    public string AssemblyName
+    {
+      get
+      {
+        return mAssemblyName;
+      }
+    }
+  }
+
   public class XmlRpcProxyGen
   {
     static Hashtable _types = new Hashtable();
@@ -51,7 +69,15 @@
         if (proxyType == null)
         {
           Guid guid = Guid.NewGuid();
-          string assemblyName = "XmlRpcProxy" + guid.ToString();
+          string assemblyName;
+
+          object[] assemblyNameAttrs = itf.GetCustomAttributes(typeof(XmlRpcProxyAssemblyNameAttribute), false);
+
+          if (assemblyNameAttrs.Length > 0)
+            assemblyName = (assemblyNameAttrs[0] as XmlRpcProxyAssemblyNameAttribute).AssemblyName;
+          else
+            assemblyName = "XmlRpcProxy" + guid.ToString();
+
           string moduleName = "XmlRpcProxy" + guid.ToString() + ".dll";
           string typeName = "XmlRpcProxy" + guid.ToString();
           AssemblyBuilder assBldr = BuildAssembly(itf, assemblyName,
